compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -241,6 +247,12 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env) {
         BLangBlockStmt queryBlock = ASTBuilderUtil.createBlockStmt(pos);
         BLangVariableReference streamRef = buildStream(clauses, queryExpr.type, env, queryBlock);
         BLangStatementExpression streamStmtExpr;
+        if (orderByClause != null) {
```

Comment:
Try to add the ballerina statements you are introducing as comments during desugaring